frontend/packages/shared/src/contexts/ServicesContext.tsx

```diff
@@ -1,27 +1,93 @@
-import React, { createContext, useContext, ReactNode } from 'react';
-import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import React, { createContext, useContext, ReactNode, useState } from 'react';
+import {
+  MutationCache,
+  MutationMeta,
+  QueryCache,
+  QueryClient,
+  QueryClientConfig,
+  QueryClientProvider,
+  QueryMeta,
+} from '@tanstack/react-query';
 import type * as queries from '../api/queries';
 import type * as mutations from '../api/mutations';
 import { ReactQueryDevtools } from '@tanstack/react-query-devtools';
+import { ToastContainer, Slide, toast } from 'react-toastify';
+import { ErrorBoundary } from 'react-error-boundary';
+import { AxiosError } from 'axios';
+import { Trans, useTranslation } from 'react-i18next';
+import { Link } from '@digdir/design-system-react';
+import { ErrorBoundaryFallback } from '../components/ErrorBoundaryFallback';
+
+import 'react-toastify/dist/ReactToastify.css';
+import 'app-shared/styles/toast.css';
 
 export type ServicesContextProps = typeof queries & typeof mutations;
 export type ServicesContextProviderProps = ServicesContextProps & {
   children?: ReactNode;
-  client?: QueryClient;
+  client?: QueryClient; // TODO : should probably be removed to force the use of QueryCache and MutationCache
+  clientConfig?: QueryClientConfig;
 };
 
 const ServicesContext = createContext<ServicesContextProps>(null);
-const queryClient = new QueryClient();
+
+const handleError = (
+  error: AxiosError,
+  t: (key: string) => string,
+  meta: QueryMeta | MutationMeta,
+): void => {
+  // TODO : log axios errors
+  // TODO : handle messages from API
+  // TODO : logout user when session is expired
+
+  if (
+    meta?.hideDefaultError === true ||
+    (meta?.hideDefaultError instanceof Function && meta?.hideDefaultError?.(error))
+  )
+    return;
+
+  toast.error(() => (
+    <Trans i18nKey={'general.error_message'} components={{ a: <Link inverted={true}>Slack</Link> }}/>
+  ), { toastId: 'default' });
+};
+
 export const ServicesContextProvider = ({
   children,
   client,
+  clientConfig,
   ...queries
 }: ServicesContextProviderProps) => {
+  const { t } = useTranslation();
+
+  const [queryClient] = useState(
+    () => client || new QueryClient({
+      ...clientConfig,
+      queryCache: new QueryCache({
+        onError: (error: AxiosError, query) => handleError(error, t, query.options?.meta),
+      }),
+      mutationCache: new MutationCache({
+        onError: (error: AxiosError, variables, context, mutation) => handleError(error, t, mutation.options?.meta),
+      }),
+    })
+  );
```

Comment:
Har ikke sett `useState` bli brukt på denne måten før. Hva er det man oppnår her? Er det for å hindre at det blir laget nye instanser av objektene etter at de allerede er opprettet?

Reply:
Ja, det er for å hindre å lage instansen på nytt, men det ville faktisk vært mer naturlig å bruke useMemo her

Reply:
I'm not sure what is best anymore after reading [this](https://github.com/TkDodo/blog-comments/issues/29#issuecomment-1015679801), [this](https://github.com/TanStack/query/issues/5458#issuecomment-1563943887), and part of [this](https://tkdodo.eu/blog/use-state-for-one-time-initializations). Dominik, one of the active maintainers of react-query, seems to strongly recommend using `useState` over `useMemo` 🤔  I reverted back to `useState` for now as it was suggested in [this example](https://github.com/TanStack/query/discussions/5279#discussioncomment-6087393)

Reply:
Interesting reading! I agree, let's keep to `useState` 🙂